### PR TITLE
 services: update the binlog message truncation flag

### PR DESCRIPTION
**Pull request by zpencer**
The truncation flag tells us whether the msg or meta was truncated.

### CODE REVIEW

File: services/src/main/java/io/grpc/services/BinlogHelper.java [hunk 1]
```diff
@@ -548,19 +538,21 @@ static Peer socketToProto(SocketAddress address) {
...
-    byte[][] serialized;
+    byte[][] serialized = null;
     if (maxHeaderBytes > 0 && (serialized = InternalMetadata.serialize(metadata)) != null) {
       int written = 0;
       for (int i = 0; i < serialized.length && written < maxHeaderBytes; i += 2) {
         byte[] key = serialized[i];
         byte[] value = serialized[i + 1];
         if (written + key.length + value.length <= maxHeaderBytes) {
-          builder.addEntry(
+          metaBuilder.addEntry(
```
Comment (carl-mastrangelo) [inline services/src/main/java/io/grpc/services/BinlogHelper.java]: addEntryBuilder().setKey(ByteString.copyFrom(key))

File: services/src/main/java/io/grpc/services/BinlogHelper.java [hunk 2]
```diff
@@ -571,21 +563,27 @@ static Peer socketToProto(SocketAddress address) {
...
         .newBuilder()
         .setFlags(flagsForMessage(compressed))
         .setLength(message.length);
     if (maxMessageBytes > 0) {
       int desiredBytes = Math.min(maxMessageBytes, message.length);
-      builder.setData(ByteString.copyFrom(message, 0, desiredBytes));
+      msgBuilder.setData(ByteString.copyFrom(message, 0, desiredBytes));
     }
-    return builder.build();
+    entryBuilder.setMessage(msgBuilder.build());
```
Comment (carl-mastrangelo) [inline services/src/main/java/io/grpc/services/BinlogHelper.java]: s/.build()//

File: services/src/main/java/io/grpc/services/BinlogHelper.java [hunk 3]
```diff
@@ -548,19 +538,21 @@ static Peer socketToProto(SocketAddress address) {
   }
 
   @VisibleForTesting
-  static io.grpc.binarylog.Metadata metadataToProto(Metadata metadata, int maxHeaderBytes) {
-    Preconditions.checkNotNull(metadata);
-    Preconditions.checkState(maxHeaderBytes >= 0);
-    Builder builder = io.grpc.binarylog.Metadata.newBuilder();
+  static void metadataToProto(
+      GrpcLogEntry.Builder entryBuilder, Metadata metadata, int maxHeaderBytes) {
```
Comment (carl-mastrangelo) [inline services/src/main/java/io/grpc/services/BinlogHelper.java]: nit: might consider naming this something like addMetadataToProto

File: services/src/main/java/io/grpc/services/BinlogHelper.java (hunk 1 above)
Comment (zpencer) [reply services/src/main/java/io/grpc/services/BinlogHelper.java]: TIL

File: services/src/main/java/io/grpc/services/BinlogHelper.java (hunk 2 above)
Comment (zpencer) [reply services/src/main/java/io/grpc/services/BinlogHelper.java]: Removed unneeded build()

File: services/src/main/java/io/grpc/services/BinlogHelper.java (hunk 3 above)
Comment (zpencer) [reply services/src/main/java/io/grpc/services/BinlogHelper.java]: renamed

**Pull request merged**